### ISSUE TITLE
localOption with TestPattern doesnt appear to work

### ISSUE THREAD

**Issue by jneira** [feuerbach/tasty#235]
Hi, i've tried to set a test pattern at compile time:

```haskell
main :: IO ()
main = do
  let withSkippedTests = Test.Tasty.localOption (error "Nay" :: TestPattern)
  Test.Tasty.defaultMain $ withSkippedTests allTests
```

This example doesnt throw the error and the entire test suite is executed.
In the example i've used error to check if option is evaluated but obviously set a correct pattern doesn't work neither.

Sorry if i am missing something basic and thanks in advance.

**Comment by feuerbach** [comment]
Yeah, this is because the pattern is one of the "global" options, and those are not taken from the test tree. You can set it using `setEnv` as described here: https://github.com/feuerbach/tasty#compile-time

That said, this is a frequent point of confusion, so I'll change the behavior to make your code work when I have time.

**Comment by feuerbach** [comment]
This has been fixed by @dcastro in #279
